lib/abstract_importer/strategies/insert_strategy.rb

```diff
@@ -53,7 +52,13 @@ def flush
 
 
       def insert_batch(batch)
-        collection.scope.insert_many(batch, @insert_options)
+        result = collection.scope.insert_many(batch, @insert_options)
+        return unless remap_ids?
+
+        map = result.each_with_object({}) do |attrs, map|
+          map[attrs.fetch("legacy_id")] = attrs.fetch("id")
+        end
+        id_map.merge! collection.table_name, map
```

Comment:
This feels like it shouldn't belong in `insert_batch`. Maybe...
```
def insert_batch(batch)
  result = collection.scope.insert_many(batch, @insert_options)
  remap_da_ids!(result) if remap_ids?
end

Reply:
👍 I could get on board with that.

Reply:
I inlined `id_map_record_batch` because I didn't like that it was a separate step in `process_record` — or that `process_record` would have to now know about the return value of `insert_many`; but I like your idea of keeping it as a separate method that is called from `insert_batch` in order to keep `insert_batch`'s logic at a single level of abstraction.